src/bin/enclave-runc-wrapper.rs

```diff
@@ -0,0 +1,241 @@
+            "--bundle" => skip_arg = true,
+            "--log" => skip_arg = true,
+            "--log-format" => skip_arg = true,
+            "--pid-file" => skip_arg = true,
+            // We want to explicitly store the value of the --root option.
+            "--root" => root_arg = true,
+            _ => {}
+        }
+        if arg.as_str().starts_with('-') {
+            // Skip options (other than --root). Those options should not be
+            // followed by a positional argument/value.
+            continue;
+        }
+
+        if skip_arg {
+            skip_arg = false;
+            continue;
+        }
+        if root_arg {
+            container_root = Some(arg.clone());
+            root_arg = false;
+            continue;
+        }
+        if container_id_arg {
+            container_id_o = Some(arg.clone());
+            container_id_arg = false;
+            continue;
+        }
+
+        match arg.as_str() {
+            "checkpoint" => container_id_arg = true,
+            "create" => {
+                container_id_arg = true;
+                add_container = true;
+            }
+            "delete" => {
+                container_id_arg = true;
+                delete_container = true;
+            }
+            "events" => container_id_arg = true,
+            "exec" => container_id_arg = true,
+            "kill" => container_id_arg = true,
+            "pause" => container_id_arg = true,
+            "ps" => container_id_arg = true,
+            "restore" => container_id_arg = true,
+            "resume" => container_id_arg = true,
+            "run" => container_id_arg = true,
+            "start" => {
+                container_id_arg = true;
+                start_container = true;
+            }
+            "state" => container_id_arg = true,
+            "update" => container_id_arg = true,
+            _ => {}
+        }
+    }
+
+    if add_container {
+        let container_key = enclave::hash(&container_id_o.unwrap())?;
+        // Initialize the container with the baseline policy.
+        enclave::add_container(
+            container_key,
+            pid_u,
+            enclave::ContainerPolicyLevel::Baseline,
+        )?;
+        cmd.status().await?;
+    } else if delete_container {
+        let container_key = enclave::hash(&container_id_o.unwrap())?;
+        enclave::delete_container(container_key)?;
+        cmd.status().await?;
+    } else if start_container {
+        let pid_u = u32::try_from(libc::pid_t::from(pid))?;
+
+        let container_id = container_id_o.unwrap();
+        let container_key = enclave::hash(&container_id)?;
+        enclave::add_process(container_key, pid_u)?;
+
+        let namespace = container_namespace(&container_id, container_root)?;
+
+        // Apply the appriopriate policy.
+        let policy = policy_label(namespace).await?;
+        enclave::write_policy(&container_id, policy)?;
+
+        cmd.status().await?;
+    } else {
```

Comment:
nit: Could be a good use for creating an enum for `*_container` and pattern matching. Depends on how you want to do this.

Reply:
Good idea